src/mlpack/core/optimizers/adam/nadamax_update.hpp

```diff
@@ -0,0 +1,167 @@
+ * Adamax and NAG to the gradient descent to improve its Performance.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_ADAM_NADAMAX_UPDATE_HPP
+#define MLPACK_CORE_OPTIMIZERS_ADAM_NADAMAX_UPDATE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * NadaMax is an optimizer that combines the AdaMax and NAG.
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @misc{Dozat2015,
+ *   title       = {},
+ *   author      = {},
+ *   institution = {},
+ *   address     = {},
+ *   year        = {}
+ * }
+ * @endcode
+ */
+class NadaMaxUpdate
+{
+ public:
+  /**
+   * Construct the Nadam update policy with the given parameters.
+   *
+   * @param epsilon The epsilon value used to initialise the squared gradient
+   *        parameter.
+   * @param beta1 The smoothing parameter.
+   * @param beta2 The second moment coefficient
+   * @param scheduleDecay The decay parameter for decay coefficients
+   */
+  NadaMaxUpdate(const double epsilon = 1e-8,
+              const double beta1 = 0.9,
+              const double beta2 = 0.99,
+              const double scheduleDecay = 4e-3)
+      :epsilon(epsilon),
+      beta1(beta1),
+      beta2(beta2),
+      scheduleDecay(scheduleDecay),
+      iteration(0),
+      cumBeta1(1)
+  {
+    // Nothing to do.
+  }
+
+  /**
+   * The Initialize method is called by SGD Optimizer method before the start 
+   * of the iteration update process.
+   *
+   * @param rows Number of rows in the gradient matrix.
+   * @param cols Number of columns in the gradient matrix.
+   */
+  void Initialize(const size_t rows, const size_t cols)
+  {
+    m = arma::zeros<arma::mat>(rows, cols);
+    u = arma::zeros<arma::mat>(rows, cols);
+  }
+
+  /**
+   * Update step for NadaMax.
+   *
+   * @param iterate Parameters that minimize the function.
+   * @param stepSize Step size to be used for the given iteration.
+   * @param gradient The gradient matrix.
+   */
+  void Update(arma::mat& iterate,
+              const double stepSize,
+              const arma::mat& gradient)
+  {
+    // Increment the iteration counter variable.
+    ++iteration;
+
+    // And update the iterate.
+    m *= beta1;
+    m += (1 - beta1) * gradient;
+
+    u *= beta2;
+    u = arma::max(u, arma::abs(gradient));
+
+    double beta1T = beta1 * (1 - (0.5 *
+        std::pow(0.96, iteration * scheduleDecay)));
+
+    double beta1T1 = beta1 * (1 - (0.5 *
+        std::pow(0.96, (iteration + 1) * scheduleDecay)));
+
+    cumBeta1 *= beta1T;
+
+    const double biasCorrection1 = 1.0 - cumBeta1;
+
+    const double biasCorrection2 = 1.0 - (cumBeta1 * beta1T1);
+
+    if ((biasCorrection1 != 0) && (biasCorrection2 != 0))
```

Comment:
Can you use ``{`` around the if condition to improve the readability?

Reply:
I have put the braces { after the condition. Let me know if you were telling about the condition and not what I have done.

Reply:
That's what I meant, thanks. I will check the update formula one more time, and I think afterwards this is ready.

Reply:
Thanks